### PR TITLE
NIFI-13198 Update RouteText not to write to FlowFiles for auto-terminated relationships

### PR DESCRIPTION
**Pull request by jrsteinebrey**
# Summary

[NIFI-13198](https://issues.apache.org/jira/browse/NIFI-13198)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI-13198) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-13198`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-13198`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Ran unit tests.
Tested the RouteText processor in the UI.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 21

### Licensing

No new files or dependencies

### Documentation

No documentation changes

### CODE REVIEW

File: nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RouteText.java [hunk 1]
```diff
@@ -549,7 +550,9 @@ public void process(final InputStream in) throws IOException {
         FlowFile flowFile = originalFlowFile;
         logger.info("Routing {} to {}", flowFile, REL_ORIGINAL);
         session.getProvenanceReporter().route(originalFlowFile, REL_ORIGINAL);
-        flowFile = session.putAttribute(flowFile, ROUTE_ATTRIBUTE_KEY, REL_ORIGINAL.getName());
+        if (!context.isAutoTerminated(REL_ORIGINAL)) {
+            flowFile = session.putAttribute(flowFile, ROUTE_ATTRIBUTE_KEY, REL_ORIGINAL.getName());
+        }
```
Comment (exceptionfactory) [inline nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RouteText.java:555]: I recommend removing this check and only retaining the conditional for the content-based processing.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
@jrsteinebrey Reviewing this change in light of the Processor design, it does not seem necessary. The `relationship != null` check already avoids unnecessary work, and it is very unlikely that the Processor would be configured to auto-terminate custom relationships. For this reason, I recommend avoid making changes to this Processor.

**Comment by exceptionfactory** [comment]
To qualify my previous comment, this could be useful in some specific scenarios, when dealing with the content specifically, so removing the conditional around the attribute change and retaining the conditional around the content handling might be useful.

File: nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RouteText.java (hunk 1 above)
Comment (jrsteinebrey) [reply nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RouteText.java]: @exceptionfactory I removed the isAutoTerminated check for REL_ORIGINAL 
and retained the conditional for the content-based processing.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the change @jrsteinebrey, the latest version looks good. +1 merging

**Pull request closed without merging**